pkg/config/config.go

```diff
@@ -26,7 +26,7 @@ func LoadFile(filename string, cfg interface{}) error {
 
 func LoadData(data []byte, cfg interface{}) error {
 	// Remove comment lines starting with #.
-	data = regexp.MustCompile(`(^|\n)\s*#.*?\n`).ReplaceAll(data, nil)
+	data = regexp.MustCompile(`(^|\n)\s*#[^\n]*`).ReplaceAll(data, nil)
```

Comment:
We don't have direct tests, but there are some indirect in pkg/mgrconfig/testdata/* which have some comments. Please add this pattern there.